src/track/globaltrackcache.cpp

```diff
@@ -419,177 +520,271 @@ GlobalTrackCacheResolver GlobalTrackCache::resolve(
                 << trackRef;
         return cacheResolver;
```

Comment:
I think we can get rid of this cacheResolver and the hard to track, very long locked period of the cache if we rely on the smart pointer here. 
It looks like it is uses to lock from the call of resolve() to initTrackId. But is there a need to lock them? 
Isn't it perfectly legal to have a track with no ID from the DB? 
Can't we lock the cache again when inserting it?

Reply:
At first sight everything seems not that complicated ;) But believe me, I worked on this for a long time and ran into lots of issues before getting it working. It bugs me that i didn't discover the tricky use after free bug early that slipped into the beta.